include/fmt/core.h

```diff
@@ -1332,7 +1332,7 @@ using ulong_type = conditional_t<long_short, unsigned, unsigned long long>;
 template <typename T> struct format_as_result {
   template <typename U,
             FMT_ENABLE_IF(std::is_enum<U>::value || std::is_class<U>::value)>
-  static auto map(U*) -> decltype(format_as(std::declval<U>()));
+  static auto map(U*) -> typename std::decay<decltype(format_as(std::declval<U>()))>::type;
```

Comment:
I think this should be `remove_cvref_t` and not `decay`.

Reply:
yes. it's a better fit in this case.